.github/workflows/test.yaml

```diff
@@ -7,9 +7,6 @@ on:
     branches: ["master"]
   merge_group:
 
-env:
-  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
```

Comment:
Just curious, do you have documentation you can link about rust-toolchain managing the crates.io lookup?  I was under the impression that rust-toolchain would handle the initial toolchain download and install, but this is referring to grabbing the crates later with cargo update.  I could easily be wrong, but I couldn't find any relevant rust-toolchain docs in a little bit of googling.

Reply:
True the documentation is vague in saying "good defaults" but there are some pr's which show this. https://github.com/dtolnay/rust-toolchain/pull/27 https://github.com/dtolnay/rust-toolchain/pull/66 https://github.com/dtolnay/rust-toolchain/pull/54

Reply:
That's good enough for me.  BTW, it looks like all of those environmental variables are also set at the top of tools/ci.sh.  Do we want to remove them there as well?